src/cdk/tree/control/nested-tree-control.ts

```diff
@@ -26,15 +26,17 @@ export class NestedTreeControl<T> extends BaseTreeControl<T> {
   expandAll(): void {
     this.expansionModel.clear();
     let toBeExpanded = <any>[];
-    this.dataNodes.forEach(dataNode => toBeExpanded.push(...this.getDescendants(dataNode)));
+    this.dataNodes.forEach(dataNode =>
```

Comment:
Rather than using a `forEach` and a `push`, you should be able to use `map` or `reduce` instead.